pkg/sql/types/types.proto

```diff
@@ -421,15 +421,12 @@ message GeoMetadata {
 // that will be serialized to disk, unlike other user defined type metadata
 // that is only stored in memory once a type is resolved.
 message PersistentUserDefinedTypeMetadata {
-  // StableTypeID is the ID of the backing TypeDescriptor for this type.
-  optional uint32 stable_type_id = 1
-    [(gogoproto.nullable) = false, (gogoproto.customname) = "StableTypeID"];
-
-  // StableArrayTypeID is the ID of the TypeDescriptor that defines the backs
-  // the automatically generated array type for this type. StableArrayTypeID
+  // ArrayTypeOID is the OID of the array type for this user defined type. It
   // is only set for user defined types that aren't arrays.
-  optional uint32 stable_array_type_id = 2
-    [(gogoproto.nullable) = false, (gogoproto.customname) = "StableArrayTypeID"];
+  optional uint32 array_type_oid = 2
+    [(gogoproto.nullable) = false, (gogoproto.customname) = "ArrayTypeOID", (gogoproto.customtype) = "github.com/lib/pq/oid.Oid"];
```

Comment:
why can we not use `oid+1` for array types in this circumstance?

Reply:
We had a small discussion about this on slack -- it's simpler when we have to remap ID's (like in backup/restore) to physically store the array type OID, rather than rely on an implicit +1

Reply:
i mean it is hidden by abstraction right? not too strongly opinioned either way, i just think seems more in line with the cleanup if we gave it some mapping treatment here too.

Reply:
The abstraction only works here because we can easily recover a descriptor ID from an OID. It would be trickier to do that when we are remapping descriptor ID's in backup and restore. I also don't think this is too unreasonable -- `pg_catalog.pg_type` stores the array type OID in a type's row entry.

Reply:
if we're certain this is the only thing stored in this meta struct, maybe we could flatten it again, haha

Reply:
(can be a separate PR though, and if we know for certain there's nothing else...)